firecracker/microvm.py

```diff
@@ -434,22 +407,7 @@ async def teardown(self):
         if self.stderr_task:
             self.stderr_task.cancel()
 
-        if self.network_tap:
-            await asyncio.sleep(
-                0.01
-            )  # Used to prevent `ioctl(TUNSETIFF): Device or resource busy`
-            logger.debug(f"Removing interface {self.network_tap}")
-            system(f"ip tuntap del {self.network_tap} mode tap")
-            logger.debug("Removing iptables rules")
-            system(
-                f"iptables -t nat -D POSTROUTING -o {self.network_interface} -j MASQUERADE"
-            )
-            system(
-                "iptables -D FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT"
-            )
-            system(
-                f"iptables -D FORWARD -i {self.network_tap} -o {self.network_interface} -j ACCEPT"
-            )
+        await Network.remove_tap_interface(self.vm_id)
```

Comment:
We may want to store a reference to this function instead of importing it, removing circular imports between packages.